src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -1424,12 +1471,42 @@ private static Set<String> validateTableProperties(Set<String> properties, Strin
         Set<String> diff = Sets.difference(lowerCaseProperties, TableAttributes.allKeywords());
 
         if (!diff.isEmpty())
-            throw new IllegalArgumentException(format("Invalid value for %s: '%s' do not parse as valid table properties",
-                                                      name, diff));
+            throw new IllegalArgumentException(invalidValueMessage(name, diff, "table"));
 
         return lowerCaseProperties;
     }
 
+    private static Set<String> validateKeyspaceProperties(Set<String> properties, String name)
+    {
+        if (properties == null)
+            throw new IllegalArgumentException(format("Invalid value for %s: null is not allowed", name));
+
+        Set<String> lowerCaseProperties = properties.stream().map(LocalizeString::toLowerCaseLocalized).collect(toSet());
+
+        for (String requiredKeyword : KeyspaceAttributes.requiredKeywords())
+        {
+            if (lowerCaseProperties.contains(requiredKeyword))
+                throw new IllegalArgumentException(format("Invalid value for %s: %s is a required keyspace property", name, Set.of(requiredKeyword)));
```

Comment:
```suggestion
                throw new IllegalArgumentException(format("Invalid value for %s: '%s' is a required keyspace property", name, Set.of(requiredKeyword)));
```
Unfortunately, would require small change in `testConfigValidation()`

Reply:
Good catch! Yes will update.